### DISCUSSION TITLE
Understanding muscle moment arm w.r.t. joint represented by quaternion

### DISCUSSION THREAD

**Discussion by zoufaond** [google-deepmind/mujoco#1785] [General]
Hello,

I tried to understand how the moment arms are calculated for quaternion. For a joint represented by euler angles the gradient of muscle lengths ($\partial l/ \partial q$) gives you moment arms acting on the generalized coordinates. I didn't understand from documention how this is done when the quaternion represents the joint as now the velocities are the angular velocities and not generalized velicoties. Or if there is some kind of paper dealing with this problem, it would be really helpful.

Thank you all.

**Comment by Balint-H** [comment]
As far as muscles acting over ball joints, the moment arm information is in the relevant tendon's Jacobian. Find the appropriate rows/columns for your ball joint in the Jacobian matrix (3 fields per tendon), which will define the moment arms with respect to the dof axes (local frame, if I understood correctly).
